testng-core/src/main/java/org/testng/reporters/MethodInvocationKey.java

```diff
@@ -0,0 +1,42 @@
+package org.testng.reporters;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+class MethodInvocationKey {
+  private final Object methodIdentity;
+  private final Object[] parameters;
+  private final int currentInvocationCount;
+
+  MethodInvocationKey(Object methodIdentity, Object[] parameters, int currentInvocationCount) {
+    this.methodIdentity = methodIdentity;
```

Comment:
```suggestion
  MethodInvocationKey(ITestNGMethod method, Object[] parameters, int currentInvocationCount) {
    this.methodIdentity = IInstanceIdentity.getInstanceId(method);
```

Reply:
I went back to my old implementation, because this suggested change is causing more failures.